lib/src/storage.cairo

```diff
@@ -0,0 +1,175 @@
+use array::ArrayTrait;
+use array::SpanTrait;
+use hash::LegacyHash;
+use serde::Serde;
+use traits::Into;
+
+#[derive(Drop)]
+struct StorageKey {
+    keys: Array<felt252>,
+    partition: felt252,
+}
+
+trait StorageKeyTrait {
+    fn new(partition: felt252, keys: Array<felt252>) -> StorageKey;
+    fn id(self: @StorageKey) -> felt252;
+}
+
+impl StorageKeyImpl of StorageKeyTrait {
+    fn new(partition: felt252, keys: Array<felt252>) -> StorageKey {
```

Comment:
good call. i think i prefer to have partition first since it follows the logical memory layout (i.e. partition -> key), so will switch the struct member order. what do you think?

Reply:
perfect